src/rust/inetstack/protocols/ethernet2/protocol.rs

```diff
@@ -1,26 +1,39 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT license.
 
+//======================================================================================================================
+// Imports
+//======================================================================================================================
+
 use crate::runtime::fail::Fail;
-use ::libc::ENOTSUP;
-use ::num_traits::FromPrimitive;
 use ::std::convert::TryFrom;
 
+//======================================================================================================================
+// Structures
+//======================================================================================================================
+
 #[repr(u16)]
-#[derive(FromPrimitive, Copy, Clone, PartialEq, Eq, Debug)]
+#[derive(Copy, Clone, PartialEq, Eq, Debug)]
 pub enum EtherType2 {
     Arp = 0x806,
     Ipv4 = 0x800,
     Ipv6 = 0x86dd,
 }
 
+//======================================================================================================================
+// Trait Implementations
+//======================================================================================================================
+
 impl TryFrom<u16> for EtherType2 {
     type Error = Fail;
 
-    fn try_from(n: u16) -> Result<Self, Fail> {
-        match FromPrimitive::from_u16(n) {
-            Some(n) => Ok(n),
-            None => Err(Fail::new(ENOTSUP, "unsupported ETHERTYPE")),
+    /// Attempts to convert a [u16] into an [EtherType2].
+    fn try_from(value: u16) -> Result<Self, Self::Error> {
+        match value {
+            0x806 => Ok(EtherType2::Arp),
```

Comment:
I assume that these constants are defined in a RFC somewhere. It might be good to define it as a const and a reference to the RFC, rather than just having these bare numbers.

Reply:
Tese are defined in the enum itself, but I cannot access them in here. The function is precisely doing the conversion. Let me know if you know a different way of doing that in pure rust. The crates used before provide this "facility" at cost of non-standard 3rd party.

Reply:
I just meant do a list of constants at the beginning of file and then use the constants in the code with a link to the enum or RFC.

Reply:
I agree.  Defining them as local constants, with a comment describing the authoritative source, would be good.